### PR TITLE
Mark several C macros imported from <float.h> as deprecated.

### PR DESCRIPTION
**Pull request by stephentyrone**
These macros all have straightforward replacements in terms of static properties on FloatingPoint or BinaryFloatingPoint. It is necessary to add 1 in a few places because of differences between how C and Swift count significand bits and normalize the significand, but this is expected to have minimal impact on code in practice (and when it does have impact, using the Swift definition is generally simpler).
rdar://problem/27871465

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci Please test

File: stdlib/public/Platform/Darwin.swift.gyb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+public let M_SQRT2 = 2.squareRoot()
+
+@available(swift, deprecated: 3.0, message: "Please use '0.5.squareRoot()'.")
+public let M_SQRT1_2 = 0.5.squareRoot()
+
+//  Constants defined by <float.h>
+@available(swift, deprecated: 3.0, message: "Please use 'T.radix' to get the radix of a FloatingPoint type 'T'.")
+public let FLT_RADIX = Double.radix
+
+%for type in ["Float","Double","Float80"]:
```
Comment (moiseev) [inline stdlib/public/Platform/Darwin.swift.gyb]: May I suggest:

```python
for type, prefix in [('Float', 'FLT'), ('Double', 'DBL'), ('Float80', 'LDBL')]:
```

File: stdlib/public/Platform/Darwin.swift.gyb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+%  implicit = ""
+#if arch(i386) || arch(x86_64)
+% end
+//  Where does the 1 come from? C counts the usually-implicit leading
+//  significand bit, whereas Swift does not. Neither is really right or wrong.
+@available(swift, deprecated: 3.0, message: "Please use '${type}.significandBitCount + 1'.")
+public let ${prefix}_MANT_DIG = ${type}.significandBitCount + 1
+
+//  Where does the 1 come from? C models floating-point numbers as having a
+//  significand in [0.5, 1), but Swift (follwing IEEE 754) considers the
```
Comment (moiseev) [inline stdlib/public/Platform/Darwin.swift.gyb]: `s/follwing/following`

File: stdlib/public/Platform/Darwin.swift.gyb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+@available(swift, deprecated: 3.0, message: "Please use '0.5.squareRoot()'.")
+public let M_SQRT1_2 = 0.5.squareRoot()
+
+//  Constants defined by <float.h>
+@available(swift, deprecated: 3.0, message: "Please use 'T.radix' to get the radix of a FloatingPoint type 'T'.")
+public let FLT_RADIX = Double.radix
+
+%for type in ["Float","Double","Float80"]:
+% prefix = {"Float":"FLT","Double":"DBL","Float80":"LDBL"}[type]
+% implicit = "implicit "
```
Comment (moiseev) [inline stdlib/public/Platform/Darwin.swift.gyb]: I don't see `${implicit}` being used anywhere. Does it belong in the comment?

File: test/stdlib/FloatConstants.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import Darwin
+#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android)
+import Glibc
+#endif
+
+_ = FLT_RADIX // expected-warning {{is deprecated}}
+
+_ =  FLT_MANT_DIG // expected-warning {{is deprecated}}
+_ =  DBL_MANT_DIG // expected-warning {{is deprecated}}
+_ = LDBL_MANT_DIG // expected-warning {{is deprecated}}
```
Comment (moiseev) [inline test/stdlib/FloatConstants.swift]: Shouldn't it be surrounded by `#if arch(i386) || arch(x86_64) ... #endif`?

**Review by moiseev — CHANGES_REQUESTED** [review]
A few nitpicks. Overall a :shipit:.

**Comment by moiseev** [comment]
Also, could you please mention the radar (<rdar://problem/27871465>) in the commit message?

**Comment by stephentyrone** [comment]
@swift-ci please test

**Review by moiseev — APPROVED** [review]

**Comment by gparker42** [comment]
Reverting in #6817. The FloatConstants test is failing on iOS. 

The test is incorrectly written for architectures that aren't i386 or x86_64; the verifier still expects the long double warnings and (correctly) doesn't get them.

**Pull request merged**